test/test.js

```diff
@@ -59,10 +59,16 @@ describe('directoryTree', () => {
 	it('should not swallow exceptions thrown in the callback function', () => {
 		const error = new Error('Something happened!');
 		const badFunction = function () {
-			dirtree('./test/test_data', {extensions:['.txt']}, function(item) {
+			dirtree('./test/test_data', {includeOnly:/\.txt$/}, function(item) {
 			  throw error;
 			});
 		}
 		expect(badFunction).to.throw(error);
 	})
+
+	it('should exclude the correct folders', () => {
+		const tree = dirtree('./test/test_data',{exclude: /another_dir/});
+		expect(tree).to.deep.equal(excludeTree);
+	});
+
```

Comment:
Let's also add a test that does both include and exclude and checks the precedence too.